Updates supported network interface controllers (#54432)
openshift/openshift-docs: modules/nw-sriov-hwol-supported-devices.adoc

|MT28880 Family [ConnectX&#8209;5{nbsp}Ex]
|15b3
|1019
|===

.Technology preview network interface controllers
[cols="1,2,1,1"]

> |===

Reviewer: |===
|Manufacturer |Model |Vendor ID | Device ID

|Mellanox 

> |MT2894 Family [ConnectX-6 Lx] support

Reviewer: I would remove "support" at the end for consistency with the other table.

|15b3
|101f

|Mellanox

> |MT42822 BlueField-2 in ConnectX-6 NIC mode support

Reviewer: I would remove "support" at the end for consistency with the other table.

|MT28880 Family [ConnectX&#8209;5{nbsp}Ex]
|15b3
|1019
|===

> .Technology preview network interface controllers

Reviewer 2: ```suggestion
.Technology Preview network interface controllers
```

[cols="1,2,1,1"]
|===
|Manufacturer |Model |Vendor ID | Device ID

|Mellanox 
|MT2894 Family [ConnectX-6 Lx]
|15b3
|101f

|Mellanox
|MT42822 BlueField-2 in ConnectX-6 NIC mode

> |15B3

Reviewer: I think we should use lower case letters here for consistency.

|a2d6
|===

> :FeatureName: ConnectX-6 Lx support and BlueField-2 in ConnectX-6 NIC mode support

Reviewer 2: Per the SSG we usually try to avoid using "support" with tech preview features. 

> Never use the phrase "supported as a Technology Preview", and avoid using "support" in Technology Preview descriptions. Instead, use neutral words like "available", "provide", "capability", and so on.

Could be rephrased, maybe like:

```suggestion
:FeatureName: Using a ConnectX-6 Lx or BlueField-2 in ConnectX-6 NIC mode device
```
